src/subdomains/payment-request/services/payment-request.service.ts

```diff
@@ -0,0 +1,129 @@
+import { PaymentRequestEntity, PaymentRequestMethod, PaymentRequestState } from '../entities/payment-request.entity';
+import { PaymentRequestRepository } from '../repositories/payment-request.repository';
+
+@Injectable()
+export class PaymentRequestService {
+  private readonly logger = new LightningLogger(PaymentRequestService);
+
+  constructor(
+    private readonly paymentRequestRepository: PaymentRequestRepository,
+    private readonly assetService: AssetService,
+  ) {}
+
+  @Cron(CronExpression.EVERY_10_SECONDS)
+  @Lock()
+  async processOpenTransactions(): Promise<void> {
+    if (Config.processDisabled(Process.UPDATE_PAYMENT_REQUEST)) return;
+
+    const pendingPaymentRequestEntities = await this.paymentRequestRepository.findBy({
+      state: PaymentRequestState.PENDING,
+    });
+
+    const currentDate = new Date();
+
+    for (const pendingPaymentRequestEntity of pendingPaymentRequestEntities) {
+      const secondsDiff = Util.secondsDiff(pendingPaymentRequestEntity.expiryDate, currentDate);
+      const timeoutWithDelay = Config.payment.timeout + Config.payment.timeoutDelay;
+
+      if (secondsDiff > timeoutWithDelay) {
+        await this.expirePaymentRequest(pendingPaymentRequestEntity);
+      }
+    }
+  }
+
+  // --- FUNCTIONS --- //
+
+  async getPaymentRequest(id: number): Promise<PaymentRequestEntity> {
+    const paymentRequest = await this.paymentRequestRepository.findOneBy({ id });
+    if (!paymentRequest) throw new NotFoundException(`Payment request with id ${id} not found`);
+
+    return paymentRequest;
+  }
+
+  async findPending(
+    transferAmount: number,
+    paymentMethod: PaymentRequestMethod,
+  ): Promise<PaymentRequestEntity | undefined> {
+    const paymentRequests = await this.paymentRequestRepository.findPending(transferAmount, paymentMethod);
+    if (!paymentRequests?.length) return;
+
+    if (1 != paymentRequests.length) {
+      for (const paymentRequest of paymentRequests) {
+        const errorMessage = `Payment request id ${paymentRequest.id}: ${paymentRequests.length} pending payment request entries with amount ${transferAmount} found`;
+        await this.failPaymentRequest(paymentRequest, errorMessage);
+      }
+
+      return;
+    }
+
+    return paymentRequests[0];
+  }
+
+  async checkDuplicate(walletPaymentParam: LightingWalletPaymentParamDto) {
+    const accountAsset = await this.assetService.getAccountAssetByNameOrThrow(walletPaymentParam.currencyCode ?? '');
+
+    const duplicates = await this.paymentRequestRepository.exist({
+      where: {
+        state: PaymentRequestState.PENDING,
+        transferAmount: Number(walletPaymentParam.amount),
+        invoiceAsset: { id: accountAsset.id },
+      },
+    });
+
+    if (duplicates)
+      throw new ServiceUnavailableException(
+        `Duplicate currency ${walletPaymentParam.currencyCode} and amount ${walletPaymentParam.amount}`,
+      );
+  }

```

Comment:
OK, let's discuss it - give me a call ;-)

Reply:
`findPending` is used, if a payment is done - therefore it uses the `transferAmount`. `checkDuplicate` is used in case of a payment request, there is no `transferAmount` available at this time. Therefore it uses now the `invoiceAmount`.